models/ROMS_rutgers/model_mod.f90

```diff
@@ -0,0 +1,1716 @@
+real(r8)    :: tops, bots            ! Ensemble level values
+real(r8)    :: bath                  ! Bathymetry at a single point
+integer(i8) :: dartidx     
+
+levels = 0
+
+do i = 1, Nc
+   bath = h(lon_c(i), lat_c(i))
+  
+   ! Physical depth is a function of SSH and differs 
+   ! across the ensemble
+   do ie = 1, ens_size
+      call compute_physical_depth(bath, SSHcorn(i, ie), depths) 
+      call depth_bounds(lon_lat_vrt(3), depths, lev(ie, 1), lev(ie, 2), lev_frc(ie), dstatus)
+     
+      if (dstatus /= 0) return
+   enddo
+
+   ! Now, get the state. First, find unique vertical levels 
+   ! used in the ensemble of vertical interpolation
+   ! This is to avoid computing/calling the state for different 
+   ! members at the same level. So, we only get the state 
+   ! ensemble at each of the interpolating levels once!
+   call unique_levels(lev(:, 1), lev(:, 2), ens_size, levels, Nl)
+
+   ! Get the state at all requested depths
+   do il = 1, Nl
+      lz = levels(il)
+      dartidx      = get_dart_vector_index(lon_c(i), lat_c(i), lz, domid, varid)
+      Zvals(lz, :) = get_state(dartidx, state_handle)
+
+      !write(*, '(A, i3, A, i3, A, i12, A, i2, A, 3F10.6)') & 
+      !         '  il: ', il, ', lvl: ', lz, ', DART indx: ', dartidx, ', Zvals(', lz, ', :): ', Zvals(lz, :)
+   enddo
+
+   ! Do the vertical interpolation using the 
+   ! state at all depth array (i.e., Zvals)
+   do ie = 1, ens_size
+      bots = Zvals(lev(ie, 1), ie)
+      tops = Zvals(lev(ie, 2), ie)
+
+      corners(i, ie) = bots + lev_frc(ie)*(tops-bots)
+   enddo
+enddo
+
+end subroutine vert_interp
+
+
+!------------------------------------------------------------------
+! Any model specific distance calcualtion can be done here
+
+subroutine get_close_obs(gc, base_loc, base_type, locs, loc_qtys, loc_types, &
+                         num_close, close_ind, dist, ens_handle)
+
+type(get_close_type),          intent(in)    :: gc            ! handle to a get_close structure
+integer,                       intent(in)    :: base_type     ! observation TYPE
+type(location_type),           intent(inout) :: base_loc      ! location of interest
+type(location_type),           intent(inout) :: locs(:)       ! obs locations
+integer,                       intent(in)    :: loc_qtys(:)   ! QTYS for obs
+integer,                       intent(in)    :: loc_types(:)  ! TYPES for obs
+integer,                       intent(out)   :: num_close     ! how many are close
+integer,                       intent(out)   :: close_ind(:)  ! incidies into the locs array
+real(r8),            optional, intent(out)   :: dist(:)       ! distances in radians
+type(ensemble_type), optional, intent(in)    :: ens_handle
+
+character(len=*), parameter :: routine = 'get_close_obs'
+
+call loc_get_close_obs(gc, base_loc, base_type, locs, loc_qtys, loc_types, &
+                          num_close, close_ind, dist, ens_handle)
+
+end subroutine get_close_obs
```

Comment:
This is just the location_mod get close so you can pass through the location_mod get_close_obs

```suggestion
```